federation-js/src/composition/validate/postComposition/keysMatchBaseService.ts

```diff
@@ -50,13 +52,17 @@ export const keysMatchBaseService: PostCompositionValidator = function ({
           .filter(([service]) => service !== serviceName)
           .forEach(([extendingService, keyFields = []]) => {
             // Extensions can't specify more than one key
+            const extendingServiceTypeNode = findTypeNodeInServiceList(parentTypeName, extendingService, serviceList);
             if (keyFields.length > 1) {
               errors.push(
                 errorWithCode(
                   'MULTIPLE_KEYS_ON_EXTENSION',
                   logServiceAndType(extendingService, parentTypeName) +
                     `is extended from service ${serviceName} but specifies multiple @key directives. Extensions may only specify one @key.`,
-                  parentType.astNode ?? undefined,
+                  extendingServiceTypeNode && 'directives' in extendingServiceTypeNode ?
+                    extendingServiceTypeNode.directives?.find(directive =>
+                      directive.name.value === 'key')?.arguments?.find
+                        (argument => argument.name.value === 'fields') : undefined,
```

Comment:
For this one, since we can't select the general "directive area" I am going to select the type node, so the error can be located to the whole error where there are too many keys.